spec/models/category_spec.rb

```diff
@@ -0,0 +1,33 @@
+require 'rails_helper'
+
+RSpec.describe Category, type: :model do
+  describe "Associations" do
+    it {is_expected.to belong_to(:user)}
+    it {is_expected.to have_many(:transactions).dependent(:destroy)}
+  end
+  describe "Validations" do
+    subject{FactoryBot.build(:category)}
+    context "when field name" do
+      it {is_expected.to validate_presence_of(:name)}
+      it {is_expected.to validate_length_of(:name).is_at_most(Settings.digits.length_name_max_25)}
+    end
+  end
+  describe "define enum for category_type" do
+    it {is_expected.to define_enum_for :category_type}
+  end
+  describe "Public class methods" do
+    let!(:user) {FactoryBot.create :user}
+    let!(:category_1) {FactoryBot.create :category, user_id: user.id}
+    let!(:category_2) {FactoryBot.create :category, user_id: user.id}
+    let!(:category_3) {FactoryBot.create :category, user_id: user.id}
+    subject {Category}
```

Comment:
subject làm gì e nhỉ?

Reply:
Dạ theo em tìm hiểu thì Subject chỉ được thực thi khi gọi tới nó (chứ không phải khi khai báo subject). Một instance của class đó sẽ được truyền đến mỗi example thông qua subject.
is_expected.to tương đương với expect(subject).to